spec/models/school_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe School do
+  before do
+    stub_user_info_api
+  end
+
+  describe 'associations' do
+    it 'has many classes' do
+      school = create(:school, classes: [build(:school_class), build(:school_class)])
+      expect(school.classes.size).to eq(2)
+    end
+
+    context 'when a school is destroyed' do
+      let!(:school_class) { build(:school_class, members: [build(:class_member)]) }
+      let!(:school) { create(:school, classes: [school_class]) }
+
+      it 'also destroys school classes to avoid making them invalid' do
+        expect { school.destroy! }.to change(SchoolClass, :count).by(-1)
+      end
+
+      it 'also destroys class members to avoid making them invalid' do
+        expect { school.destroy! }.to change(ClassMember, :count).by(-1)
+      end
+    end
+  end
+
+  describe 'validations' do
+    subject(:school) { build(:school) }
+
+    it 'has a valid default factory' do
+      expect(school).to be_valid
+    end
+
+    it 'can save the default factory' do
+      expect { school.save! }.not_to raise_error
+    end
+
+    # The school's ID must be set before create from the profile app's organisation ID.
+    # This avoids having two different IDs for a school which would be confusing.
```

Comment:
❓ determine efficient order of creating organisations, users and roles in profile to reference in app